File: app/src/main/java/com/example/weatherchallengemvvmkotlin/data/network/WeatherService.java
package com.example.weatherchallengemvvmkotlin.data.network;

import com.example.weatherchallengemvvmkotlin.data.models.WeatherResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherService {

    @GET("data/2.5/weather")
    Observable<WeatherResponse> getWeather(@Query("q") String query, @Query("APPID") String appId);
}

File: app/src/main/java/com/example/weatherchallengemvvmkotlin/data/network/WeatherService.kt
package com.example.weatherchallengemvvmkotlin.data.network

import com.example.weatherchallengemvvmkotlin.data.models.WeatherResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherService {
    @GET("data/2.5/weather")
    fun getWeather(@Query("q") query: String?, @Query("APPID") appId: String?): Observable<WeatherResponse?>?
}